File: app/src/main/java/uk/co/victoriajanedavis/chatapp/data/model/db/converter/BooleanConverter.java
package uk.co.victoriajanedavis.chatapp.data.model.db.converter;

import androidx.room.TypeConverter;

public class BooleanConverter {
    @TypeConverter
    public static boolean toBoolean(int value) {
        return (value != 0);
    }

    @TypeConverter
    public static int toInt(boolean value) {
        return (value ? 1 : 0);
    }

}

File: app/src/main/java/uk/co/victoriajanedavis/chatapp/data/model/db/converter/BooleanConverter.kt
package uk.co.victoriajanedavis.chatapp.data.model.db.converter

import androidx.room.TypeConverter

class BooleanConverter {
    @TypeConverter
    fun toBoolean(value: Int): Boolean {
        return value != 0
    }

    @TypeConverter
    fun toInt(value: Boolean): Int {
        return if (value) 1 else 0
    }
}
